mosquitto/2.0.15.patch

```diff
@@ -0,0 +1,148 @@
+From a840ddcf4c7c75d5dc28a8c154b51fad9a39b29e Mon Sep 17 00:00:00 2001
+From: Eric Blankenhorn <eric@wolfssl.com>
+Date: Mon, 23 Jan 2023 08:20:51 -0600
+Subject: [PATCH] Support wolfssl instead of openssl
+
+Changes:
+- `config.mk`: add the `WITH_TLS=wolfssl` option
+- The`wolfssl/options.h` header is included by defining the `EXTERNAL_OPTS_OPENVPN` macro
+- `net_mosq.c`: UI_METHOD not implemented in wolfSSL
+- `net_mosq.h`: UI_METHOD not implemented in wolfSSL
+- `net_mosq_ocsp.c`: safestack.h not implemented in wolfSSL
+
+wolfSSL
+
+```
+git clone https://github.com/wolfSSL/wolfssl.git
+cd wolfssl
+./autogen.sh
+./configure --enable-opensslextra --enable-opensslall --enable-ocsp --enable-ocspstapling --enable-context-extra-user-data --enable-psk --enable-sessioncerts --enable-crl CFLAGS="-DOPENSSL_VERSION_NUMBER=0x10100000"
```

Comment:
We should add a --enable-mosquitto to wolfSSL which enables all of these options, including the OPENSSL_VERSION_NUMBER override.  That override should happen in wolfssl/openssl/opensslv.h.  We can add a new flag for mosquitto, maybe HAVE_MOSQUITTO or WOLFSSL_MOSQUITTO which you can use for the override.

Reply:
I'm a little torn on this one. I feel like we already have too many config options! I'll bring it up in with the team today.